File: Pragmatism/Server/api/dto.go
package api

type AuthToken struct {
	Signature string `json:"signature"` // token
	Payload   string `json:"payload"`   // payload
}

type TaskSubmitRequest struct {
	UserId    string     `json:"userId"`
	TaskId    int        `json:"taskId"`
	TaskDir   string     `json:"taskDir"`
	TaskFiles []TaskFile `json:"taskFiles"`
	Language  string     `json:"language"`
}

type GetTaskResponse struct {
	TaskDir string `json:"taskDir"` // Unique identifier for the task
	// TaskFiles contains the files related to the task
	TaskFiles []TaskFile `json:"taskFiles"` // List of files related to the task
}

const (
	FILE_TYPE_LIB  = 0
	FILE_TYPE_TASK = 1
)

type TaskFile struct {
	FileName string `json:"fileName"` // Name of the file
	Content  string `json:"content"`  // Content of the file
	FileType int    `json:"fileType"`
}

type TestResult struct {
	TestName     string `json:"testName"`
	IsSuccessful bool   `json:"isSuccessful"`
}

type TaskSubmitResponse struct {
	TestResults []*TestResult `json:"testResults"`
	DbgLogs     string        `json:"dbgLogs"`
}

type HttpHandlerErrorResponse struct {
	ShouldRetry bool `json:"shouldRetry"`
}

type SolvedTask struct {
	TaskId   uint32 `json:"taskId"`
	Title    string `json:"title"`
	Solution string `json:"solution"`
	SolvedAt string `json:"solved_at"`
}

File: Pragmatism/Client/src/dto/dto.ts
export namespace DTO {
    export type Task = {
        id: number;
        title: string;
        description: string;
        numSolves: number;
        markdownUrl: string;
        taskDir: string;
    };

    export type TaskSubmitRequest = {
        userName: string;
        userId: string;
        taskId: number;
        taskDir: string;
        taskFiles: TaskFile[];
        language: string;
    };

    export enum FileType {
        LIB = 0,
        TASK = 1,
    }

    export type TaskFile = {
        fileName: string;
        content: string;
        fileType: FileType;
    };

    export type GetTaskResponse = {
        taskDir: string;
        taskFiles: TaskFile[];
        markdownUrl: string;
        language: string;
    };

    export type TestResult = {
        testName: string;
        isSuccessful: boolean;
    };

    export type TaskSubmitResponse = {
        testResults: TestResult[];
        dbgLogs: string;
    };

    export type User = {
        id: string;
        name: string;
        picture: string;
    };

    export type SolvedTask = {
        title: string;
    };
}
